source/extensions/transport_sockets/tcp_stats/tcp_stats.cc

```diff
@@ -133,6 +133,13 @@ void TcpStatsSocket::recordStats() {
     config_->stats_.cx_tx_percent_retransmitted_segments_.recordValue(percent_retransmissions);
   }
 
+  const uint64_t mss = (tcp_info->tcpi_snd_mss > 0) ? tcp_info->tcpi_snd_mss : 1460;
```

Comment:
Is a default of 1460 a valid assumption in all cases? I'd say if the kernel sets it to zero, don't record a value.